### PR TITLE
f-button@v1.0.0 - Extend f-button to include icon

### PR DESCRIPTION
**Pull request by nehadhanuka-justeat**
## UI Review Checks

- [x] README and/or UI Documentation has been [created|updated]
- [ ] Unit tests will be added in separate PR

## Browsers Tested

- [x] Chrome (latest)
- [ ] Internet Explorer 11
- [ ] Mobile (Please list device/browser – Ideally one iPhone model and one Android)

### CODE REVIEW

File: packages/components/atoms/f-button/README.md [hunk 1]
```diff
@@ -75,11 +77,12 @@ The props that can be defined are as follows:
 
 | Prop  | Type  | Required | Default | Description |
 | ----- | ----- | -------- |------- | ----------- |
-| `buttonType` | `String` | No |`primary` | Sets the modifier theme for styling.<br>Options: `primary`, `secondary`, `outline`, `ghost`, `link`, `icon`. |
+| `buttonType` | `String` | No |`primary` | Sets the modifier theme for styling.<br>Options: `primary`, `secondary`, `outline`, `ghost`, `link`. |
 | `buttonSize` | `String` | No |`medium` | Sets the size.<br>Options: `large`, `medium`, `small`, `xsmall`. |
 | `isFullWidth` | `Boolean` | No |  `false` | Controls whether or not to apply fullWidth modifier class |
 | `actionType` | `String` | No |`button` | Sets the action button type.<br>Options: `button`, `submit`, `reset`. |
 | `href` | `String` | No |  N/A | If provided, replaces `Action` button or `Link` button. |
+| `isIcon` | `Boolean` | No |  `false` | Sets the buttin style to be displayed like an icon . |
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/README.md]:
```suggestion
| `isIcon` | `Boolean` | No |  `false` | Sets the button style to be displayed like an icon . |
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 2]
```diff
@@ -114,6 +119,15 @@ $btn-sizeSmall-padding          : 7px 1em 9px;
...
 
+$icon-bgColor                   : $blue--offWhite;
+$icon-bgColor--hover            : $blue--offWhite--dark;
+$icon-bgColor--active           : $blue--offWhite--darkest;
+
+$icon-default-width             : 1.5em;
+$icon-sizeSmall-width           : 1em;
+$icon-sizeLarge-width           : 2em;
+
+$icon-default-padding           : 3px;
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue]: Doesn't look like these variables are used currently – can they be removed?

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 3]
```diff
@@ -313,22 +327,63 @@ $btn-sizeXSmall-lineHeight      : 1;
...
 }
 
+
 /**
- * Modifier – .o-btn--icon
- *
- * Removes unneeded styles from button when only an icon is on it (like a close button)
+ * ==========================================================================
+ * Icon style and modifiers
+ * ==========================================================================
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue]:
```suggestion
 * Modifier – .o-btn--icon
 *
 * Handles styling when only an icon is displayed within the button component
```

File: packages/components/atoms/f-button/src/components/Button.vue [hunk 4]
```diff
@@ -313,22 +327,63 @@ $btn-sizeXSmall-lineHeight      : 1;
...
+      }
+
+      &:active,
+      &:focus {
+          path {
+               fill: $color-link-active;
+          }
+      }
+ }
+}
```
Comment (ashleynolan) [inline packages/components/atoms/f-button/src/components/Button.vue]: Just an indentation fix:

```suggestion
    &.o-btn--primary {
        path {
            fill: $btn-primary-textColor;
        }
    }

    &.o-btn--secondary {
        path {
            fill: $btn-secondary-textColor;
        }
    }

    &.o-btn--outline {
        path {
            fill: $btn-outline-textColor;
        }
     }

    &.o-btn--ghost {
        path {
            fill: $btn-ghost-textColor;
        }
    }

    &.o-btn--link {
        path {
            fill: $color-link-default;
        }

        &:hover {
            path {
                fill: $color-link-hover;
            }
        }

        &:active,
        &:focus {
            path {
                fill: $color-link-active;
            }
        }
    }
}
```

**Review by ashleynolan — COMMENTED** [review]
Looks good – a few small comments, but otherwise 👍

**Review by ashleynolan — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

**Pull request merged**